File: SMART/app/src/main/java/com/cs565project/smart/db/entities/RecommendationActivity.java
package com.cs565project.smart.db.entities;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

@Entity
public class RecommendationActivity {
    @PrimaryKey(autoGenerate = true)
    public int id;

    public String activityName;
    public boolean isSet;
    public int timeOfDay;
    public String activityType;

    public RecommendationActivity(String activityName, boolean isSet, int timeOfDay, String activityType) {
        this.activityName = activityName;
        this.isSet = isSet;
        this.timeOfDay = timeOfDay;
        this.activityType = activityType;
    }

}

File: SMART/app/src/main/java/com/cs565project/smart/db/entities/RecommendationActivity.kt
package com.cs565project.smart.db.entities

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

/**
 * Database entity to hold details of activities which can be recommended to the user.
 */
@Entity
class RecommendationActivity(var activityName: String, var isSet: Boolean, var timeOfDay: Int, var activityType: String) {
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

}
